### PR TITLE
Fix fallout from ghc-typelits-natnormalise-0.7.5 release

### LINKED ISSUE #1861: cabal build error solveSimpleWanteds: too many iterations
I started seeing this error a day ago on a few different projects when compiling clash-prelude-1.4.2 with multiple GHC versions and cabal versions. For example when running `cabal build` with [ice40-prim](https://github.com/standardsemiconductor/ice40-prim):

```
...
src/Clash/Sized/Internal/Index.hs:1:1: error:
    solveSimpleWanteds: too many iterations (limit = 4)
      Set limit with -fconstraint-solver-iterations=n; n=0 for no limit
      Simples = {[WD] $dKnownNat_a3IvT {0}:: KnownNat n0 (CNonCanonical),
                 [WD] $dKnownNat_a3IvV {0}:: KnownNat n (CNonCanonical),
                 [WD] $dKnownNat_a3IvY {0}:: KnownNat m0 (CNonCanonical),
                 [WD] $dIntegral_a3Iw1 {0}:: Integral a0 (CNonCanonical),
                 [WD] hole{co_a3Iw2} {0}:: Index (((n - 1) * (n - 1)) + 1)
                                           ~ a0 (CNonCanonical),
                 [WD] hole{co_a3Iw3} {0}:: Index m0 ~ r0 (CNonCanonical)}
      WC = WC {wc_simple =
                 [WD] $dKnownNat_a3INU {0}:: KnownNat 2 (CNonCanonical)
                 [WD] $dKnownNat_a3INV {0}:: KnownNat 2 (CNonCanonical)}
  |
1 | {-|
  | ^
cabal: Failed to build clash-prelude-1.4.2 (which is required by
ice40-prim-0.3.1.0). See the build log above for details.
```
Update: I get the same error when building the [clash-starters simple](https://github.com/clash-lang/clash-starters/tree/main/simple) project too. OS is Ubuntu 20.04.

### PR DESCRIPTION
**Pull request by christiaanb**
Fixes #1861

### CODE REVIEW

**Comment by christiaanb** [comment]
Backporting to 1.2 as well because stackage lts-17 is still on clash-1.2

**Comment by christiaanb** [comment]
So it seems the 0.7.5 release of ghc-typelits-natnormalise is broken, so I've deprecated its release on hackage, meaning it shouldn't get picked up in any of the build plans any longer.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -87,8 +87,8 @@ haddock:
     expire_in: 1 month
   script:
     - .ci/build_docs.sh
-  # Run on shared runners
-  tags:
+  # Do not run on shared runners
+  tags: [ local ]
```
Comment (christiaanb) [inline .gitlab-ci.yml:91]: So the haddock job kept getting killed with code 137 (OOM), but the memory usage increase is somewhat hard to debug. So I hope we can all live with this.

**Comment by christiaanb** [comment]
Removed the backport, because cabal should make a build plan with the latest 0.7.6 release now.

File: .gitlab-ci.yml (hunk 1 above)
Comment (alex-mckenna) [reply .gitlab-ci.yml:91]: This exists on my CI changes branch anyway, so me and @martijnbastiaan can definitely live with it

Comment (martijnbastiaan) [reply .gitlab-ci.yml:91]: Jep

**Comment by christiaanb** [comment]
This PR is no longer needed.

**Pull request closed without merging**